File: 240411/좌우로 움직이는 로봇/robot-moving-from-side-to-side.py
N, M = map(int, input().split())
diff = N - M if N > M else M - N
robot_A, robot_B = [], []

cur_a, cur_b = 0, 0
for _ in range(N):
    size, mv_dir = input().split()
    size = int(size)

    if mv_dir == 'R':
        for i in range(cur_a, cur_a + size):
            robot_A.append(i)
        cur_a += size 
    else:
        for i in range(cur_a, cur_a - size, -1):
            robot_A.append(i)
        cur_a -= size

for _ in range(M):
    size, mv_dir = input().split()
    size = int(size)

    if mv_dir == 'R':
        for i in range(cur_b, cur_b + size):
            robot_B.append(i)
        cur_b += size
    else:
        for i in range(cur_b, cur_b - size, -1):
            robot_B.append(i)
        cur_b -= size

if len(robot_A) > len(robot_B):
    for _ in range(diff):
        robot_B.append(robot_B[-1])
else:
    for _ in range(diff):
        robot_A.append(robot_A[-1])

print(robot_A)
print(robot_B)

answer = 0
for (a_elem_idx, a_elem), (b_elem_idx, b_elem) in zip(enumerate(robot_A[1:]), enumerate(robot_B[1:])):
    if a_elem == b_elem and robot_A[a_elem_idx - 1] != robot_B[b_elem_idx - 1]:
        answer += 1

print(answer)

File: 260126/좌우로 움직이는 로봇/robot-moving-from-side-to-side.cpp
#include <bits/stdc++.h>
using namespace std;


int main() 
{
    int N, M;
    cin >> N >> M;

    vector<long long> robotA, robotB;
    robotA.push_back(0);
    robotB.push_back(0);

    long long curA = 0, curB = 0;

    for (int i = 0; i < N; i++) 
    {
        long long size;
        char dir;
        cin >> size >> dir;

        if (dir == 'R') 
        {
            for (long long x = curA + 1; x <= curA + size; x++) robotA.push_back(x);
            curA += size;
        } 
        else 
        {
            for (long long x = curA - 1; x >= curA - size; x--) robotA.push_back(x);
            curA -= size;
        }
    }

    for (int i = 0; i < M; i++) 
    {
        long long size;
        char dir;
        cin >> size >> dir;

        if (dir == 'R') 
        {
            for (long long x = curB + 1; x <= curB + size; x++) robotB.push_back(x);
            curB += size;
        } 
        else 
        {
            for (long long x = curB - 1; x >= curB - size; x--) robotB.push_back(x);
            curB -= size;
        }
    }

    if (robotA.size() > robotB.size()) 
    {
        size_t diff = robotA.size() - robotB.size();
        robotB.insert(robotB.end(), diff, robotB.back());
    } 
    else if (robotB.size() > robotA.size()) 
    {
        size_t diff = robotB.size() - robotA.size();
        robotA.insert(robotA.end(), diff, robotA.back());
    }

    long long answer = 0;
    for (size_t i = 1; i < robotA.size(); i++) 
    {
        if (robotA[i] == robotB[i] && robotA[i - 1] != robotB[i - 1]) 
        {
            answer++;
        }
    }

    cout << answer << "\n";
    return 0;
}